run/markdown-preview/renderer/main.js

```diff
@@ -0,0 +1,50 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const express = require('express');
+const MarkdownIt = require('markdown-it');
+const xss = require('xss');
+
+const app = express();
+
+app.use(express.json());
+
+app.post('/', (req, res) => {
+  try {
+
+    // Get the Markdown text and convert it into HTML using markdown-it
+    const markdown = req.body.markdown;
```

Comment:
Let's add a check that this exists

Reply:
@averikitsch what would be best practice if body.markdown doesn't exist? Throw an error?

Reply:
You do have this wrapped in a try/catch block, so you could throw an error just to be caught down below. Or we could think about making the try/catch block more specific, because you probably want to translate the error to the users like, "Markdown could not be converted to HTML". With that said, we should log an error and send back a 400 response

Reply:
@averikitsch Let me know if this looks good: [latest commit](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/pull/1752/commits/a8952faf515a52a70319efcf17e15b9be8284ca7)